handlers/deposit_file.go

```diff
@@ -39,37 +38,46 @@ func (d *depositFileEntry) Handle(params operations.DepositFileParams) middlewar
 		panic(err)
 	}
 
-	location, err := d.copyFileToStorage(id, params.Upload)
+	upload := d.paramsToFile(params)
+	location, err := d.copyFileToStorage(id, upload)
 	if err != nil {
 		panic(err)
 	}
 
 	log.Printf("The location of the file is: %s", *location)
 
-	if err := d.createFileResource(id, params.Upload.Header.Filename); err != nil {
+	if err := d.createFileResource(id, upload.Metadata); err != nil {
 		panic(err)
 	}
 	// TODO: return file location: https://github.com/sul-dlss-labs/taco/issues/160
 	response := map[string]interface{}{"id": id}
 	return operations.NewDepositResourceCreated().WithPayload(response)
 }
 
-func (d *depositFileEntry) copyFileToStorage(id string, file runtime.File) (*string, error) {
-	filename := file.Header.Filename
-	contentType := file.Header.Header.Get("Content-Type")
-	log.Printf("Saving file \"%s\" with content-type: %s", filename, contentType)
+func (d *depositFileEntry) paramsToFile(params operations.DepositFileParams) *uploaded.File {
+	file := params.Upload
+	fileHeader := file.Header
+	metadata := uploaded.FileMetadata{
+		Filename:    fileHeader.Filename,
+		ContentType: fileHeader.Header.Get("Content-Type"),
```

Comment:
add header variable for `FilesetID` - either here or second PR. We expect all Files deposited will know the Fileset that contains them

Reply:
That's certainly something we should do, but it was not in the scope of #267.

Reply:
because #267 was written before consulting / reviewing what we needed for `File` metadata. It is, however, a requirement for our expectations of any `File` metadata resource.

Reply:
Huh, didn't this come out of our team meeting on Feb 28th? I thought this is where we talked about writing that ticket.

Reply:
Created a ticket to do this work: https://github.com/sul-dlss-labs/taco/issues/376